server/controllers/userController.js

```diff
@@ -602,219 +238,28 @@ app.post('/account', [
-    };      
-    const managementRval = await pool.query(managementAccountQuery);
-    var managed = false
-    if(managementRval.rows.length > 0){
-      for(r of managementRval.rows){
-        if(r.parent_entity_id == entityId){
-          managed = true;
-          break;
-        }
-      }
-    }
-
-    if(managed == false){
-      res.status(401).json({
-        message:"You do not manage the sender of this transfer"
-      });
-      return;
-    }
-  }
-
-
-  if(receiverEntityId != entityId){
-    // check if this is a valid subaccount
-    const managementAccountQuery = {
-      text: `SELECT *
-      FROM entity_manager
-      WHERE child_entity_id = $1
-      AND entity_manager.active = TRUE`,
-      values: [receiverEntityId]
-    };      
-    const managementRval = await pool.query(managementAccountQuery);
-    var managed = false
-    if(managementRval.rows.length > 0){
-      for(r of managementRval.rows){
-        if(r.parent_entity_id == entityId){
-          managed = true;
-          break;
-        }
-      }
-    }
-
-    if(managed == false){
-      res.status(401).json({
-        message:"You do not manage the receiver of this transfer"
-      });
-      return;
-    }
-  }
-
-
-
-  // Find tokens for this bundle
-  const queryTokens = {
-    text: `SELECT *
-    FROM token
-    WHERE entity_id = $1
-    ORDER BY id ASC
-    LIMIT $2`,
-    values: [senderEntityId, bundleSize]
-  }
-  const rvalTokens = await pool.query(queryTokens);
-  const tokens = rvalTokens.rows; 
-  
-  if(tokens.length != bundleSize){
-    res.status(422).json({
-      message:"Not enough tokens matching bundle description"
-    });
-    return;
-  }
-
-  const tokenUUIDs = [];
-  for(token of tokens){
-    tokenUUIDs.push(token.uuid);
-  }
-
-
-  // move tokens
-  const query2 = {
-    text: `UPDATE token
-    SET entity_id = $1
-    WHERE uuid = ANY ($2)`,
-    values : [receiverEntityId, tokenUUIDs]
-  }
-  const rval2 = await pool.query(query2);
-
-  const response = {
-    status: `${tokens.length} tokens transferred to ${receiverWallet}`,
-    wallet_url: config.wallet_url + "?wallet="+receiverWallet
-  }
-  res.status(200).json(response);
-  res.end();
-
-}));
-
-
-
-app.post('/transfer', [
-
-    check('tokens[*].*').isUUID(),
-
-    check('sender_wallet', 'Invalid Sender wallet name').isAlphanumeric(),
-
-    check('receiver_wallet', 'Invalid Reciever wallet name').isAlphanumeric()
-    
-], asyncHandler(async (req, res, next) => {
-
-  const errors = validationResult(req);
-
-  if (!errors.isEmpty()) {
-     return res.status(422).json({ errors: errors.array() });
-  }
-
-  const entityId = req.entity_id;
-  const accessGranted = await checkAccess(entityId, 'manage_accounts');
-  if( !accessGranted ){
-    res.status(401).json([{
-      msg:"Not Permitted",
-      param: "manage accounts",
-      location:"access_control"
-    }]);
-    return;
-  }
+  const entityId = res.locals.entity_id;
+  assert(entityId);
```

Comment:
assert throws an exception - I'm not sure what gets returned to the API client in the case of an exception.   Instead the inputs to the controller function should pre-screened.  If no entity is found, the API can return a 401 to the client and never enter the userController, since an authenticated entityId (wallet) is required to use all paths except the authentication path itself.

Reply:
Hm I actually don't think we need the asserts on the entityId given that for each of these routes, the entity_id is supplied  by and stored on res.locals by the verifyJWT middleware called right before it. So if none were supplied, the global error handler should already pick that up within verifyJWT and we would never enter this middleware as you pointed out. And each of the routes that access this entityId require the user to be logged in, so the verifyJWT will be called before that. 

I agree on rewriting the asserts on the uuid for the supplied token on the get / token/:uuid to check if it's supplied on the parameter. Just to clarify, since I didn't see this specific route on the postman docs, do we want to return a server error when no token is passed in as a params or is there something else we should return?

Reply:
I added some 'assert', sorry about that, actually, I just add if to help develop and debug, javascript's weak type makes it too flexible and unexpectable, so some assert check would make dev easier I think.

Reply:
Yeah can definitely help in dev mode. I went ahead and just removed the asserts for now (since those asserts on the entityId are not seeded by our database anyway and is returned from an earlier middleware in the chain) and am returning a 401 if no token is supplied in req.params in the get / token/:uuid route.

Reply:
No problem!